### ISSUE TITLE
Overarching Idea - The Start of Documentation

### ISSUE THREAD

**Issue by Adron** [ThrashingCode/colligere#6]
Alright, so here's the idea (I'll turn this into a blog entry too, but starting here as just an issue)...

This will be a simple CLI App that can be used to send generated data against and API end point. I'm thinking, something along the lines of the schema to generate from (like columns, key value, data types, etc) would be a file (or a default something maybe?), passed in to the command like this, with a URI passed in for the simplist of execution. Kind of like this.

`colligere deluge` would be the root command with the following parameters/switches passed to the command.

`uri="http://api.someplace.com/dataingest"` for the URI to send the requests against.

`verb="get"` for specifying the http verb to use, post, etc.

`schema="/the/file/location/relative/to/cli/executing/thefile.json"` wherever the schema file is that is to be used to generate the data that will be submitted to the uri end point.

What am I missing at this point that needs to be sent against the uri?

@dbriesz @jinglish @olsonso

**Comment by dbriesz** [comment]
Should we have a parameter/flag for a token/authentication too?

**Comment by Adron** [comment]
Yeah, I think that would be important to have at some point, or some way to set a file or environment variables to pass whatever type of security keys, username and passwords, or whatever is setup for the API the data is being sent to. The only question is, what's the easiest way to do this so that it's super super super easy.  :)

**Comment by jinglish** [comment]
I'd second a file- or envvar-based to solution--I've seen CLI apps that require secrets as arguments before, and I'm not a fan.

**Comment by Adron** [comment]
Alright. I'm going to knock this out and implement some basic commands so we have a skeleton base to work with. I'm going to follow the standard Cobra CLI to implement it per how I wrote about [here](https://compositecode.blog/2018/01/31/building-a-data-thrashing-cli-tool-in-go/). Will close this once I implement those basic commands and open up issues per each command and the respective documentation we'll need around that so that this "getting started" doc I wrote is closed and we can break out future documentation needs independently.
